src/Paket/Program.fs

```diff
@@ -90,106 +79,68 @@ with
+        | None -> args
+
+    let rest = 
+        if verbose then rest |> List.filter (fun a -> a <> "-v" && a <> "--verbose")
+        else rest
+
+    verbose, logFile, rest
+
+let showHelp (helpTopic:HelpTexts.CommandHelpTopic) = 
+                        tracefn "%s" helpTopic.Title
+                        tracefn "%s" helpTopic.Text
+
+let v, logFile, args = filterGlobalArgs (Environment.GetCommandLineArgs() |> Seq.skip 1 |> Seq.toList)
+
+let commandArgs<'T when 'T :> IArgParserTemplate> args = 
+    UnionArgParser
+        .Create<'T>()
+        .Parse(inputs = Array.ofList args, raiseOnUsage = false, errorHandler = ProcessExiter())
+
+Logging.verbose <- v
+Option.iter setLogFile logFile
 
 try
-    match results with
-    | Some(command,results) ->
-        let force = results.Contains <@ CLIArguments.Force @> 
-        let interactive = results.Contains <@ CLIArguments.Interactive @> 
-        let hard = results.Contains <@ CLIArguments.Hard @> 
-        let noInstall = results.Contains <@ CLIArguments.No_Install @>
-        let noAutoRestore = results.Contains <@ CLIArguments.No_Auto_Restore @>
-        let includePrereleases = results.Contains <@ CLIArguments.Include_Prereleases @> 
-        let withBindingRedirects = results.Contains <@ CLIArguments.Redirects @>
-
-        if results.IsUsageRequested then 
-            let showHelp (helpTopic:HelpTexts.CommandHelpTopic) = 
-                tracefn "%s" helpTopic.Title
-                tracefn "%s" helpTopic.Text
-
-            match command with
-            | Command.Init -> showHelp HelpTexts.commands.["init"]
-            | Command.Add -> showHelp HelpTexts.commands.["add"]
-            | Command.Remove -> showHelp HelpTexts.commands.["remove"]
-            | Command.Install -> showHelp HelpTexts.commands.["install"]
-            | Command.Restore -> showHelp HelpTexts.commands.["restore"]
-            | Command.Update -> showHelp HelpTexts.commands.["update"]
-            | Command.Outdated -> showHelp HelpTexts.commands.["outdated"]
-            | Command.InitAutoRestore -> showHelp HelpTexts.commands.["init-auto-restore"]
-            | Command.ConvertFromNuget -> showHelp HelpTexts.commands.["convert-from-nuget"]
-            | Command.Simplify -> showHelp HelpTexts.commands.["simplify"]
-            | Command.FindRefs -> showHelp HelpTexts.commands.["find-refs"]
-            | Command.Config -> showHelp HelpTexts.commands.["config"]
-            | Command.Unknown -> traceErrorfn "no command given.%s" (parser.Usage())
-
-        else
-            match command with
-            | Command.Init -> Dependencies.Init()
-            | Command.Add -> 
-                let packageName = results.GetResult <@ CLIArguments.Nuget @>
-                let version = 
-                    match results.TryGetResult <@ CLIArguments.Version @> with
-                    | Some x -> x
-                    | _ -> ""
+    match args with
+        | "add" :: args ->
```

Comment:
Actually neither version looks good to me.
The old one do not support our use case for having separate argument for each command well.
The new one tries to provide the desired behavior but introduces redundancy and magic values.

I have to look further into `Nessos.UnionArgParser` to see how we can do better here.
I'll try to look at it on this weekend.